src/System.Console/src/System/IO/SyncTextReader.cs

```diff
@@ -31,8 +31,8 @@ protected override void Dispose(bool disposing)
             {
                 lock (_methodLock)
                 {
-                    // Explicitly pick up a potentially methodimpl'ed Dispose
-                    ((IDisposable)_in).Dispose();
+                    // This will always pick up TextReader.Dispose
```

Comment:
@James-Ko, it's a situation like this:
```C#
using System;
using System.IO;
using System.Text;

class Program
{
    static void Main()
    {
        TextWriter writer = new MyTextWriter();
        ((IDisposable)writer).Dispose();
        writer.Dispose();
    }
}

class MyTextWriter : TextWriter, IDisposable
{
    void IDisposable.Dispose()
    {
        Console.WriteLine("IDisposable.Dispose()");
    }

    public override Encoding Encoding { get { return Encoding.UTF8; } }
}
```
Run that, and you'll see the output message only output for the call via ((IDisposable)writer).Dispose, so the point @nguerrera was raising is that if such an instance was passed in, changing how we invoke Dispose could stop invoking that custom Dispose implementation.